quil-rs/src/parser/lexer/mod.rs

```diff
@@ -262,21 +262,17 @@ fn lex_identifier_raw(input: LexInput) -> InternalLexResult<String> {
         map(
             tuple::<_, _, InternalLexError, _>((
                 take_while1(is_valid_identifier_leading_character),
-                take_while(is_valid_identifier_middle_character),
+                take_while(is_valid_identifier_end_character),
+                recognize(many0(pair(
+                    take_while1(is_dash),
+                    take_while1(is_valid_identifier_end_character),
+                ))),
             )),
-            |(left, right)| format!("{left}{right}"),
+            |(leading, middle, trailing_dash_vars)| {
+                format!("{leading}{middle}{trailing_dash_vars}")
+            },
```

Comment:
The idea here is to parse any valid identifier up to any dashes. After that point, any dash should be followed by at least one valid end character for an identifier. Any such pairs are returned and appended to the resulting identifier.

Reply:
> After that point, any dash should be followed by at least one valid end character for an identifier

"any dash" or "any dash sequence"? If not the latter, then https://github.com/rigetti/quil-rs/pull/286/files#diff-a5bcc7a39fd5fe75b07e825fc96eeaa32b6cbed091f5936b09d97f9e9112769eR267 seems off

Reply:
Yeah, the latter, a singular dash or a sequence of dashes are both valid, as long a valid end character comes after.